test/entity/user.test.js

```diff
@@ -962,7 +962,7 @@ describe('User', function() {
       const user = new User({ _id: randomString(), email: randomString() });
 
       nock(this.client.apiHostname)
-        .delete(`${user.pathname}/${user._id}`)
+        .delete(`${user.pathname}/${user._id}?hard=false`)
```

Comment:
Ideally, for a parameter that's not required by the backend, we shouldn't send a value unless the user specifically provided it.